nmigen_boards/arty_a7.py

```diff
@@ -0,0 +1,252 @@
+        Resource("user_btn", 3, Pins("B8", dir="i"), Attrs(IOSTANDARD="LVCMOS33")),
+
+        Resource("serial", 0,
+            Subsignal("tx", Pins("D10", dir="o")),
+            Subsignal("rx", Pins("A9",  dir="i")),
+            Attrs(IOSTANDARD="LVCMOS33")
+        ),
+
+        Resource("cpu_reset", 0, Pins("C2", dir="o"), Attrs(IOSTANDARD="LVCMOS33")),
+
+        Resource("spi", 0,
+            Subsignal("clk",  Pins("F1",  dir="o")),
+            Subsignal("cs",   PinsN("C1", dir="o")),
+            Subsignal("mosi", Pins("H1",  dir="o")),
+            Subsignal("miso", Pins("G1",  dir="i")),
+            Attrs(IOSTANDARD="LVCMOS33")
+        ),
+
+        Resource("i2c", 0,
+            Subsignal("scl",        Pins("L18", dir="io")),
+            Subsignal("sda",        Pins("M18", dir="io")),
+            Subsignal("scl_pullup", Pins("A14", dir="o")),
+            Subsignal("sda_pullup", Pins("A13", dir="o")),
+            Attrs(IOSTANDARD="LVCMOS33")
+        ),
+
+        *SPIFlashResources(0,
+            cs="L13", clk="L16", mosi="K17", miso="K18", wp="L14", hold="M14",
+            attrs=Attrs(IOSTANDARD="LVCMOS33")
+        ),
+
+        Resource("ddram", 0,
+            Subsignal("a",
+                Pins("R2 M6 N4 T1 N6 R7 V6 U7 R8 V7 R6 U6 T6 T8", dir="o"),
+                Attrs(IOSTANDARD="SSTL135")
+            ),
+            Subsignal("ba",  Pins("R1 P4 P2", dir="o"), Attrs(IOSTANDARD="SSTL135")),
+            Subsignal("ras", PinsN("P3",      dir="o"), Attrs(IOSTANDARD="SSTL135")),
+            Subsignal("cas", PinsN("M4",      dir="o"), Attrs(IOSTANDARD="SSTL135")),
+            Subsignal("we",  PinsN("P5",      dir="o"), Attrs(IOSTANDARD="SSTL135")),
+            Subsignal("cs",  PinsN("U8",      dir="o"), Attrs(IOSTANDARD="SSTL135")),
+            Subsignal("dm",  Pins("L1 U1",    dir="o"), Attrs(IOSTANDARD="SSTL135")),
+            Subsignal("dq",
+                Pins("K5 L3 K3 L6 M3 M1 L4 M2 V4 T5 U4 V5 V1 T3 U3 R3", dir="io"),
+                Attrs(IOSTANDARD="SSTL135", IN_TERM="UNTUNED_SPLIT_40")
+            ),
+            Subsignal("dqs",
+                DiffPairs("N2 U2", "N1 V2", dir="io"),
+                Attrs(IOSTANDARD="DIFF_SSTL135")
+            ),
+            Subsignal("clk",
+                DiffPairs("U9", "V9", dir="o"),
+                Attrs(IOSTANDARD="DIFF_SSTL135")
+            ),
+            Subsignal("cke",   Pins("N5",  dir="o"), Attrs(IOSTANDARD="SSTL135")),
+            Subsignal("odt",   Pins("R5",  dir="o"), Attrs(IOSTANDARD="SSTL135")),
+            Subsignal("reset", PinsN("K6", dir="o"), Attrs(IOSTANDARD="SSTL135")),
+            Attrs(SLEW="FAST"),
+        ),
+
+        Resource("eth_ref_clk", 0, Pins("G18", dir="o"), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("eth_clocks", 0,
+            Subsignal("tx", Pins("H16", dir="i")),
+            Subsignal("rx", Pins("F15", dir="i")),
+            Attrs(IOSTANDARD="LVCMOS33")
+        ),
```

Comment:
I think these clocks should be a part of the `eth_mii` resource. (That's what I've done for Versa ECP5).